src/runtime/impure/lfortran_intrinsics.c

```diff
@@ -649,12 +649,20 @@ LFORTRAN_API char* _lfortran_str_copy(char* s, int32_t idx1, int32_t idx2) {
     if(idx2 <= 0) {
         idx2 = s_len + idx2;
     }
-    char* dest_char = (char*)malloc(idx2-idx1+2);
-    for (int i=idx1; i <= idx2; i++)
-    {
-        dest_char[i-idx1] = s[i-1];
+    char* dest_char;
+    if (idx3 > 0 && (idx1 <= idx2)) {
+        dest_char = (char*)malloc(idx2-idx1+2);
```

Comment:
Well in case `idx3` is present the length will not be `idx2 - idx1 + 2` I think. The factor `idx3` will come in. I think the correct length would be, `(idx2 - idx1)/idx3 + 1 + 2`. Using `idx2 - idx1 + 2` is wastage of memory by huge margins if `idx3` is not `1`.